File: src/main/java/indi/wookat/quickgrab/service/StatisticsService.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  indi.wookat.quickgrab.mapper.ResultsMapper
 *  indi.wookat.quickgrab.service.StatisticsService
 *  jakarta.annotation.Resource
 *  org.springframework.stereotype.Service
 */
package indi.wookat.quickgrab.service;

import indi.wookat.quickgrab.mapper.ResultsMapper;
import jakarta.annotation.Resource;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;
import org.springframework.stereotype.Service;

@Service
public class StatisticsService {
    @Resource
    private ResultsMapper resultsMapper;

    public List<Map<String, Object>> getStatistics(Integer buyerId, LocalDateTime startTime, LocalDateTime endTime) {
        return this.resultsMapper.getStatistics(buyerId, startTime, endTime);
    }

    public List<Map<String, Object>> getDailyStats(Integer buyerId, Integer status) {
        return this.resultsMapper.getDailyStats(buyerId, status);
    }

    public List<Map<String, Object>> getHourlyStats(Integer buyerId, Integer status) {
        return this.resultsMapper.getHourlyStats(buyerId, status);
    }
}

File: cpp/src/service/StatisticsService.cpp
#include "quickgrab/service/StatisticsService.hpp"

namespace quickgrab::service {

StatisticsService::StatisticsService(repository::ResultsRepository& results,
                                     repository::BuyersRepository& buyers)
    : results_(results)
    , buyers_(buyers) {}

std::vector<repository::ResultsRepository::AggregatedStats> StatisticsService::getStatistics(
    const std::optional<int>& buyerId,
    const std::optional<std::string>& startTime,
    const std::optional<std::string>& endTime) {
    return results_.getStatistics(buyerId, startTime, endTime);
}

std::vector<repository::ResultsRepository::DailyStat> StatisticsService::getDailyStats(
    const std::optional<int>& buyerId,
    const std::optional<int>& status) {
    return results_.getDailyStats(buyerId, status);
}

std::vector<repository::ResultsRepository::HourlyStat> StatisticsService::getHourlyStats(
    const std::optional<int>& buyerId,
    const std::optional<int>& status) {
    return results_.getHourlyStats(buyerId, status);
}

std::vector<model::Buyer> StatisticsService::getAllBuyers() {
    return buyers_.findAll();
}

} // namespace quickgrab::service

